src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt

```diff
@@ -86,12 +87,60 @@ class SnykLanguageClient :
     private val progressEndMsgCache: Cache<String, WorkDoneProgressEnd> =
         Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.SECONDS).build()
 
+
     override fun telemetryEvent(`object`: Any?) {
         // do nothing
     }
 
-    override fun publishDiagnostics(diagnostics: PublishDiagnosticsParams?) {
-        // do nothing for now
+    override fun publishDiagnostics(diagnosticsParams: PublishDiagnosticsParams?) {
+        if (diagnosticsParams == null) {
+            return
+        }
+
+        updateCache(diagnosticsParams)
+    }
+
+    private fun updateCache(diagnosticsParams: PublishDiagnosticsParams) {
+        val filePath = diagnosticsParams.uri
+
+        try {
+            getScanPublishersFor(filePath.toVirtualFile().path).forEach { (project, scanPublisher) ->
+                val snykFile = SnykFile(project, filePath.toVirtualFile())
+                val firstDiagnostic = diagnosticsParams.diagnostics.firstOrNull()
+                val product = firstDiagnostic?.source
+
+                //If the diagnostics for the file is empty, clear the cache.
+                if (firstDiagnostic == null) {
+                    scanPublisher.onPublishDiagnostics("code", snykFile, emptyList())
```

Comment:
I might be remembering it wrong, but didn't we have constants for that somewhere?